File: 1. Two Sum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = (nums, target) => {
  const previousValues = {};

  for (let i = 0; i < nums.length; i++) {
    let remainingSum = target - nums[i];

    if (previousValues[remainingSum] !== undefined) {
      return [previousValues[remainingSum], i];
    } else {
      previousValues[nums[i]] = i;
    }
  }
  
  return [];
};

const nums = [2, 7, 11, 15];
const target = 9;
console.log('twoSum: ', twoSum(nums, target));

File: 1. Two Sum/twoSum.py
from typing import List

class Solution:
  def twoSum(self, nums: List[int], target: int) -> List[int]:
    previousValues = {}

    for i in range(len(nums)):
      remainingSum = target - nums[i]

      if remainingSum in previousValues:
        return [previousValues[remainingSum], i]
      else:
        previousValues[nums[i]] = i

    return []
  
nums = [2, 7, 11, 15]
target = 9
solution = Solution()
print('twoSum: ', solution.twoSum(nums, target))